tests/include/drivers/keygen.h

```diff
@@ -29,11 +29,25 @@
 #if defined(PSA_CRYPTO_DRIVER_TEST)
 #include <psa/crypto_driver_common.h>
 
-extern void *test_driver_keygen_forced_output;
-extern size_t test_driver_keygen_forced_output_length;
-
-extern psa_status_t test_transparent_keygen_status;
-extern unsigned long test_transparent_keygen_hit;
+struct test_driver_keygen_hooks_s {
```

Comment:
Please define a type (`typedef … …_t`) and use it in preference to `struct xxx_s`. The fact that the type is a struct is only relevant when accessing its fields, not when referencing the type (e.g. when declaring a variable). Defining a struct name is optional: here I don't see the point.

(The reason `psa/crypto_struct.h` refers to `struct xxx_s` is that the type is defined in another file, which in turn is for a reason that doesn't apply here: the existence of the type is public and standard, but the struct layout is implementation-dependent.)

Reply:
That's how I set it up initially until I looked at `psa_structs.h`. Changed.